migration_steps/validation/api_tests/shared/api_test.py

```diff
@@ -310,28 +311,48 @@ def get_formatted_api_response(
             self.api_log(f"Checking responses from: {endpoint_final}")
 
             response_text = self.get_response_object(endpoint_final)
-            json_obj = json.loads(response_text)
+            if response_text:
+                json_obj = json.loads(response_text)
 
-            if self.list_in_field:
-                json_obj = json_obj[self.list_in_field]
+                if self.list_in_field:
+                    json_obj = json_obj[self.list_in_field]
 
-            json_items_to_loop_through = self.get_json_items_to_loop_through(json_obj)
-
-            for json_item_to_inspect in json_items_to_loop_through:
-                var_to_eval = f"json_item_to_inspect{json_locator}"
-                rationalised_var = self.rationalise_var(
-                    var_to_eval, json_item_to_inspect
+                json_items_to_loop_through = self.get_json_items_to_loop_through(
+                    json_obj
                 )
-                try:
-                    response_struct[json_locator] = (
-                        response_struct[json_locator] + rationalised_var + "|"
-                    )
-                except Exception:
-                    response_struct[json_locator] = rationalised_var + "|"
+
+                for json_item_to_inspect in json_items_to_loop_through:
+                    self.result_list = []
+                    self.get_json_block_results(json_item_to_inspect, json_locator)
+
+                    for result in self.result_list:
+                        try:
+                            response_struct[json_locator] = (
+                                response_struct[json_locator] + result + "|"
+                            )
+                        except Exception:
+                            response_struct[json_locator] = result + "|"
 
         log.debug(f"returning: {response_struct}")
         return response_struct
 
+    def get_json_block_results(self, json_item_to_inspect, json_locator):
```

Comment:
My suggestion would be to use jsonpath to apply the selector to the response JSON. That way you've got a lot more flexibility and you can remove all the code for managing wildcards in the path and hand it over to the library.